images/20-progs/Dockerfile

```diff
@@ -26,3 +26,17 @@ COPY --from=k3os /output/ /output/
 WORKDIR /output
 RUN git clone --branch v0.7.0 https://github.com/ahmetb/kubectx.git \
  && chmod -v +x kubectx/kubectx kubectx/kubens
+# ETCD
+# AMD64 and ARM64 is supported, otherwise etcd is disabled
+ARG ARCH
+ENV ETCD_VER v3.4.7
+ENV ETCD_DOWNLOAD_URL https://github.com/etcd-io/etcd/releases/download
+RUN mkdir etcd
+RUN if [ "$ARCH" == "amd64" ] || [ "$ARCH" == "arm64" ]; then \
+      apk -U add curl; \
+      curl -L ${ETCD_DOWNLOAD_URL}/${ETCD_VER}/etcd-${ETCD_VER}-linux-${ARCH}.tar.gz -o ${ETCD_VER}-linux-${ARCH}.tar.gz; \
+      tar xzvf ${ETCD_VER}-linux-${ARCH}.tar.gz --strip-components=1 -C etcd; \
+      rm -f ${ETCD_VER}-linux-${ARCH}.tar.gz; \
+    else \
+      touch etcd/etcd.disabled; \
```

Comment:
I can't see that this is used somewhere to check wether to start etcd or not, am I missing something?

Reply:
In ./images/20-rootfs/Dockerfile Line 87:

COPY --from=progs /output/etcd/* /usr/src/image/sbin

Docker currently can't conditionally check if a directory is empty or present before copying, therefore etcd.disabled is added to allow for the k3os system to know that etcd is disabled as well as allow for the copy to occur if the binaries are present.

Reply:
Yes, but if I understand the PR correctly the etcd service would always be started just that it would fail on arm 32-bit and start filling up the error logs as etcd binary won't exist?

Reply:
like, or maybe the startup service checks that the etcd binary exists before starting and logging errors etc?

Reply:
If the etcd configuration is not present then the service is stopped and not added to startup. There could be a condition where a non AMD64/ARCH64 user added the etcd configuration, the system would attempt to start etcd but this would fail. Do you think an additional binary check should be conducted on the startup script to catch this?

Reply:
Aah yeah, right... completely forgot about that... 😄 Not sure, thinking that people tend to miss things in the docs and might try running it on an unsupported platform. I'll defer to @dweomer on this.